docs/integrations/destinations/s3-migrations.md

```diff
@@ -0,0 +1,202 @@
+| `{ "type": "array" }`                                                                  | `[1, "Alice"]` | `{ "type": "string" }`                                          | `"[1, \"Alice\"]"` |
+| `{ "type": "array", "items": { "oneOf": [ {"type": "integer", "type": "string"} ] } }` | `[1, "Alice"]` | `{ "type": "array", "items": [ "null", "integer", "string" ] }` | `[1, "Alice"]`     |
+| `{ "type": "array", "items": { "oneOf": [ {"type": "integer", "type": "string"} ] } }` | `[1, false]`   | `{ "type": "array", "items": [ "null", "integer", "string" ] }` | `[1, null*]`       |
+
+*The nulled fields represent conversion failures and will appear in `_airbyte_meta.changes[]`.
+
+This behavior will be applied to both Avro and Parquet formats.
+
+### Union Types (Parquet Only)
+
+#### Disjoint record improvements
+
+Formerly, unions in parquet were represented as anonymous disjoint records. For example:
+
+| Airbyte Union Type                                                                                                                     | Old Parquet Type                                                                                                         |
+|----------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------|
+| `{"oneOf": [ {"type": "integer"}, {"type": "string"} ] }`                                                                              | `Optional Record { member0: Optional Int32, member1: Optional String }`                                                  |
+| `{"oneOf": [ {"type": "boolean"}, {"type": "object", "properties": { "id": { "type": "integer" }, "name": { "type": string" } } } ] }` | `Optional Record { member0: Optional Boolean, member1: Optional Record { id: Optional Uint32, name: Optional String } }` |
+
+Now unions will be represented as typed disjoint records with named fields. For example:
+
+| Airbyte Union Type                                                                                                                     | New Parquet Type                                                                                                                               |
+|----------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------|
+| `{"oneOf": [ {"type": "integer"}, {"type": "string"} ] }`                                                                              | `Optional Record { type: Optional String, integer: Optional Int32, string: Optional String }`                                                  |
```

Comment:
should we clarify that `type` is always nonnull, even though it's generated as `Optional String`?

(... can we generate the schema as non-optional string?)

Reply:
Yeah, that annoys me too. The existing schema converter makes everything nullable. There's no easy way to hack around it. I'll clarify.

Reply:
I hacked around it. :) 


```
  }, {
    "name" : "union_with_dupes",
    "type" : [ "null", {
      "type" : "record",
      "name" : "union_with_dupes",
      "namespace" : "",
      "fields" : [ {
        "name" : "type",
        "type" : "string"
      }, {
        "name" : "integer",
        "type" : [ "null", "long" ],
        "default" : null
      }, {
        "name" : "string",
        "type" : [ "null", "string" ],
        "default" : null
      } ]
    } ],
    "default" : null
  } ]
```

But the hacks are in the old schema converter, so they can go away later.

I'll add tests for it and push the fix / update this documentation tomorrow morning.

Reply:
mrrgh. this is what I was afraid of, when we started passing around jsonschema blobs between mappers  :P  but similar to everything else, I think it goes away if we have a json/avro record converter that understands AirbyteType :shrug:

Reply:
Tests added, documentation fixed